server/tests/message.integration.spec.js

```diff
@@ -259,6 +259,25 @@ describe('Message API:', () => {
             .expect(httpStatus.OK)
 
         );
+
+        it('should be archivable', () => request(app)
+            .put(`${baseURL}/message/archive/${originalMessageId}`)
```

Comment:
You may want to configure this test and the full sequence of tests so that they don't depend on previous tests. Right now, this test will pass if the message referred to by `originalMessageId` was somehow already archived. In tests using async behavior we have to be very careful about interactions between tests.